### ISSUE TITLE
zip-builtin-not-iterating (W1637) is triggered on Python 3

### ISSUE THREAD

**Issue by belgiancoder** [PyCQA/pylint#2651]
### Steps to reproduce
This code (simple csv plotter) triggers W1637 in Python 3.7.0
```
import csv
import matplotlib.pyplot as plt

with open(f'{name}.csv') as file:
        reader = csv.reader(file, delimiter=';')
        col = next(reader)
        plt.plot(*zip(*reader))

plt.show()

# 'zip()' is used in this case to transpose
# [[1,2], [3,4], [5,6]] -> [1,2,3], [4,5,6]
```
### Current behavior
The error triggers and makes no sense.

### Expected behavior
According to the doc:
>zip-builtin-not-iterating (W1637) zip built-in referenced when not iterating Used when the zip built- in is referenced in a non-iterating context (returns an iterator in Python 3) **This message can’t be emitted when using Python >= 3.0.**

So it shouldn't be possible to trigger it at all.

### Version
```
>>> pylint.__version__
'2.2.2'
```

**Comment by belgiancoder** [comment]
I use VSCode and it appears this warning was enabled. I looked at the doc why this occurs and saw it `can’t be emitted when using Python >= 3.0` so I reported the bug. Either the error is in the doc, or it is a false trigger. But either case it needs fixing. As it isn't my computer, I don't want to mess with config files

**Comment by PCManticore** [comment]
@belgiancoder Thanks for the comment. It's definitely a documentation problem as well, as the `can’t be emitted when using Python >= 3.0` part needs to be removed. But it's definitely not a false trigger as well, the message is currently enabled in your configuration file somehow and it's going to be emitted every time pylint encounters those constructs.

**Comment by PCManticore** [comment]
Awesome, glad that's settled. On a second look, I think there's also a bug here as that unpacking's definitely an iterating context.

So for context this check was emitted on Python 2 with the Python 3 porting checker. In Python 2, `zip` returned a list, which means that a lot of operations that are invalid in Python 3 were done on the output of `zip`, such as indexing elements and such, so that's what the check is supposed to find, operations against a `zip` that is not in an iterable, materializing context. But since pylint 2.X moved to Python 3 only, we still look for these checks, if they are activated with `--py3k` as that is our signal that we are linting against a Python 2 file. But the bug here is that `*` is definitely a materializing context, so the check shouldn't apply to it.

**Comment by belgiancoder** [comment]
So this **IS** a false trigger after all **AND** a documentation issue as well. Do I get a bonus point ?
